truss/contexts/image_builder/serving_image_builder.py

```diff
@@ -198,85 +293,46 @@ def update_model_name(config: TrussConfig, model_key: str) -> str:
             "Key for model missing in config or incorrect key used. Use `model` for VLLM and `model_id` for TGI."
         )
     model_name = config.build.arguments[model_key]
-    if "gs://" in model_name:
+    if "gs://" in model_name or "s3://" in model_name:
         # if we are pulling from a gs bucket, we want to alias it as a part of the cache
-        model_to_cache = HuggingFaceModel(model_name)
-        config.hf_cache.models.append(model_to_cache)
+        model_to_cache = ModelRepo(model_name)
+        config.model_cache.models.append(model_to_cache)
 
-        config.build.arguments[
-            model_key
-        ] = f"/app/hf_cache/{model_name.replace('gs://', '')}"
+        prefix_removed = model_name[4:]  # removes "gs://" or "s3://"
+
+        config.build.arguments[model_key] = f"/app/model_cache/{prefix_removed}"
     return model_name
 
 
 def get_files_to_cache(config: TrussConfig, truss_dir: Path, build_dir: Path):
     def copy_into_build_dir(from_path: Path, path_in_build_dir: str):
         copy_tree_or_file(from_path, build_dir / path_in_build_dir)  # type: ignore[operator]
 
-    model_files = {}
-    cached_files: List[str] = []
-    if config.hf_cache:
+    remote_model_files = {}
+    local_cached_files: List[CachedFile] = []
+    if config.model_cache:
         curr_dir = Path(__file__).parent.resolve()
         copy_into_build_dir(curr_dir / "cache_warmer.py", "cache_warmer.py")
-        for model in config.hf_cache.models:
+        for model in config.model_cache.models:
             repo_id = model.repo_id
             revision = model.revision
 
             allow_patterns = model.allow_patterns
             ignore_patterns = model.ignore_patterns
 
-            filtered_repo_files = list(
-                filter_repo_objects(
-                    items=list_files(
-                        repo_id, truss_dir / config.data_dir, revision=revision
-                    ),
-                    allow_patterns=allow_patterns,
-                    ignore_patterns=ignore_patterns,
-                )
-            )
-
-            cached_files = fetch_files_to_cache(
-                cached_files, repo_id, filtered_repo_files
-            )
+            model_cache = RemoteCache.from_repo(repo_id, truss_dir / config.data_dir)
+            remote_filtered_files = model_cache.filter(allow_patterns, ignore_patterns)
+            local_cached_files += model_cache.prepare_for_cache(remote_filtered_files)
```

Comment:
I think `local_cached_files` makes it seem like they are already cached. That hasn't happened yet, maybe `files_to_cache`?